File: ilamicbook/backend-fastapi/src/main/python/com/ilamicbook/backend/services/book_service.py
from typing import List
from ..models.book import Book

class BookService:
    """Service layer for book operations"""
    
    def __init__(self):
        # Mock data - in real app this would come from database
        self.books = [
            Book(
                id=1,
                title="The Great Gatsby",
                author="F. Scott Fitzgerald",
                description="A story of the fabulously wealthy Jay Gatsby and his love for the beautiful Daisy Buchanan."
            ),
            Book(
                id=2,
                title="To Kill a Mockingbird",
                author="Harper Lee",
                description="The story of young Scout Finch and her father Atticus in a racially divided Alabama town."
            ),
            Book(
                id=3,
                title="1984",
                author="George Orwell",
                description="A dystopian novel about totalitarianism and surveillance society."
            )
        ]
    
    def get_all_books(self) -> List[Book]:
        """Get all books"""
        return self.books
    
    def get_book_by_id(self, book_id: int) -> Book:
        """Get book by ID"""
        for book in self.books:
            if book.id == book_id:
                return book
        return None 

File: ilamicbook/backend-node/src/main/javascript/com/ilamicbook/backend/services/bookService.js
const Book = require('../models/book');

/**
 * Service layer for book operations
 */
class BookService {
    constructor() {
        // Mock data - in real app this would come from database
        this.books = [
            new Book(
                1,
                "The Great Gatsby",
                "F. Scott Fitzgerald",
                "A story of the fabulously wealthy Jay Gatsby and his love for the beautiful Daisy Buchanan."
            ),
            new Book(
                2,
                "To Kill a Mockingbird",
                "Harper Lee",
                "The story of young Scout Finch and her father Atticus in a racially divided Alabama town."
            ),
            new Book(
                3,
                "1984",
                "George Orwell",
                "A dystopian novel about totalitarianism and surveillance society."
            )
        ];
    }

    /**
     * Get all books
     */
    getAllBooks() {
        return this.books.map(book => book.toJSON());
    }

    /**
     * Get book by ID
     */
    getBookById(id) {
        const book = this.books.find(book => book.id === id);
        return book ? book.toJSON() : null;
    }

    /**
     * Create a new book
     */
    createBook(bookData) {
        const newBook = new Book(
            this.books.length + 1,
            bookData.title,
            bookData.author,
            bookData.description
        );
        this.books.push(newBook);
        return newBook.toJSON();
    }

    /**
     * Update a book
     */
    updateBook(id, bookData) {
        const bookIndex = this.books.findIndex(book => book.id === id);
        if (bookIndex === -1) {
            return null;
        }

        const updatedBook = new Book(
            id,
            bookData.title || this.books[bookIndex].title,
            bookData.author || this.books[bookIndex].author,
            bookData.description || this.books[bookIndex].description,
            this.books[bookIndex].createdAt,
            new Date()
        );

        this.books[bookIndex] = updatedBook;
        return updatedBook.toJSON();
    }

    /**
     * Delete a book
     */
    deleteBook(id) {
        const bookIndex = this.books.findIndex(book => book.id === id);
        if (bookIndex === -1) {
            return false;
        }

        this.books.splice(bookIndex, 1);
        return true;
    }
}

module.exports = BookService; 